.github/workflows/r-ci.yml

```diff
@@ -52,6 +53,63 @@ jobs:
       - name: Bootstrap
         run: cd apis/r && tools/r-ci.sh bootstrap
 
+      - name: Set additional repositories (macOS)
+        if: ${{ matrix.os != 'ubuntu-latest' }}
+        run: echo 'options(repos = c("https://tiledb-inc.r-universe.dev", getOption("repos")))' | tee -a ~/.Rprofile
+
+      - name: Set additional repositories (Linux)
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: |
+          rversion <- paste(strsplit(as.character(getRversion()), split = '\\.')[[1L]][1:2], collapse = '.')
+          codename <-  system('. /etc/os-release; echo ${VERSION_CODENAME}', intern = TRUE)
+          repo <- "https://tiledb-inc.r-universe.dev"
+          (opt <- sprintf('options(repos = c("%s/bin/linux/%s/%s", "%s", getOption("repos")))', repo, codename, rversion, repo))
+          cat(opt, "\n", file = "~/.Rprofile", append = TRUE)
```

Comment:
This won't work, as I just discovered.  r-universe already jumped to noble whereas ubuntu-latest will be jammy for some time.  

![image](https://github.com/single-cell-data/TileDB-SOMA/assets/673121/ff9b30fe-c97c-4989-9a9e-bb83fe28d676)

So we are once again between a rock and a hard place.  Once r2u jumps to 0.27.0 (by default tomorrow we're in trouble and the above won't save us.  :disappointed:

Reply:
That's why I've added the source-only repo as an additional repo https://github.com/single-cell-data/TileDB-SOMA/actions/runs/9022913531/job/24793545830?pr=2540#step:8:1; if possible, CI prefers a binary install but if none matches the constrains, we have the source to fallback on
![image](https://github.com/single-cell-data/TileDB-SOMA/assets/11035068/717ca39a-7b62-46fd-a511-3740e8b335d4)

Reply:
I fear you may need to also deal with `bspm` to re-enable source and deal with the sorting.  I am sure you can sort that out tomorrow when the need arises...

Reply:
And there is simply no linux/jammy/4.4 -- I wish there were. (You can hit eg the URL https://tiledb-inc.r-universe.dev/bin/linux/noble/4.4/src/contrib/PACKAGES in the browser -- this one corresponds to one of the builds from my screenshot above -- but other combos such as jammy/4.4 failed for me and likely do not exist.

We're still between a rock and a hard place.  So maybe starting tomorrow source builds for tiledb-r until SOMA flips to the newer one.  We'll survive...

Reply:
When doing the actual install, I do the `options(bspm.version.check = TRUE)` to allow bspm to search elsewhere https://github.com/single-cell-data/TileDB-SOMA/pull/2540/files#diff-5331055c5808c739bfd30c63013fd35ce1e28e3edef1df4476550617d393467cR72; in addition, I tell `install.packages()` to use the specific repository for tiledb-r that has the valid version https://github.com/single-cell-data/TileDB-SOMA/pull/2540/files#diff-5331055c5808c739bfd30c63013fd35ce1e28e3edef1df4476550617d393467cR109-R110

I'm happy to hold off merging this until tomorrow when r2u gets 0.27.0 and re-run CI to test

Reply:
Merge as you see fit, this is mostly a heads-up that I would not be surprised to see new and different breakage tomorrow.

Which can be dealt with tomorrow, I suppose.